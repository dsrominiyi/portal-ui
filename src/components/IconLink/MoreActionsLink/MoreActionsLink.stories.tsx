import React from 'react';
import { storiesOf } from '@storybook/react';

import IconLinkAction from '.';

const stories = storiesOf('MoreActionsLink', module);

const render = (onClick: () => void) => (props?: any) => (
  <IconLinkAction key={props && props.key} onClick={onClick} />
);

export const storyRenders = {
  moreIcon: render(() => alert('You clicked!'))
};

stories.add('With on click', storyRenders.moreIcon);
