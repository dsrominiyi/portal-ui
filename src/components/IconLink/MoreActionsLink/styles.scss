@import '../../../style/vars';

.icon-link-action {
  position: relative;
  transition: 0.1s ease-in-out;
  transition-property: color, background-color, border-color;

  &::before {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    right: 100%;
    border-bottom: 1px solid #333;
    transition: right 0.3s ease-out;
  }

  &:focus::before,
  &:hover::before {
    right: 0;
  }
}
