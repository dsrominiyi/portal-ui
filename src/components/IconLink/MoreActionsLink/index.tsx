import React, { FunctionComponent, MouseEvent } from 'react';
import MoreIcon from 'uikit-icons/lib/More';
import IconLink from '../index';
import styles from './styles.scss';

interface Props {
  onClick(event: MouseEvent): void;
}

export const MoreActionsLink: FunctionComponent<Props> = ({ onClick }) => (
  <IconLink
    icon={<MoreIcon onClick={onClick} />}
    additionalClasses={[styles.iconLinkAction]}
    title="More"
  />
);

export default MoreActionsLink;
