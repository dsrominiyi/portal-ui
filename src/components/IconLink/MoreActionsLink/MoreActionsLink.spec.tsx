import { storyRenders } from './MoreActionsLink.stories';

import testStorySnapshots from '../../../helpers/test/testStorySnapshots';

describe('IconLinkAction', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
