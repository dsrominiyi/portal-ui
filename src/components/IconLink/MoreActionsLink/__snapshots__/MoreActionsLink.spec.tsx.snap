// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLinkAction component story renders match snapshots: moreIcon 1`] = `
<IconLink
  additionalClasses={
    Array [
      "iconLinkAction",
    ]
  }
  icon={
    <SvgMore
      onClick={[Function]}
    />
  }
  title="More"
/>
`;
